test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/CollectionTests/Server/CollectionNullableFacetTestController.cs

```diff
@@ -0,0 +1,56 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="CollectionNullableFacetTestController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.Default;
+
+namespace Microsoft.OData.Client.E2E.Tests.CollectionTests.Server
+{
+    public class CollectionNullableFacetTestController : ODataController
+    {
+        [EnableQuery]
+        [HttpGet("odata/Customers")]
+        public IActionResult GetCustomers()
+        {
+            var rowIndex = DefaultDataSource.Customers;
+            return Ok(rowIndex);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/Customers({key})")]
+        public IActionResult GetCustomer([FromRoute] int key)

```

Comment:
Not a must-fix: Is it possible to convert this method and the rest to `async`
```suggestion
        public async Task<IActionResult> GetCustomerAsync([FromRoute] int key)

```

Reply:
Is there any advantage of having async methods here. I do not have any IO or compute intensive logic here. The data here is not from the database also it's static data.